test/integration-test/src/utils.rs

```diff
@@ -0,0 +1,74 @@
+//! Utilities to run tests
+
+use std::{
+    fs::File,
+    io::Write as _,
+    os::fd::AsRawFd as _,
+    process::{Command, Stdio},
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use libc::{setns, CLONE_NEWNET};
+
+fn script(script: &str) {
+    // We pipe stdout/err to capture it and print it ourselves, so that it gets captured by the
+    // test harness
+    let mut bash = Command::new("/bin/bash")
```

Comment:
why do we need to shell out to bash rather than use `Command::new("ip", ...)` multiple times?

Reply:
Pure convenience, and let's be honest, laziness.

This allows to write setup scripts without repeating the `Command::new` boilerplate code every time. Since this is made to do some setup, it allows to make readable setup scripts. Here only to create the network namespace, but it could be used by a test to create veths, install IPs, etc. And having 6 invocations of `Command::new(..).args(..).output().unwrap();` is just unwieldy.

This comes out of what we do at work (granted for actual applications and not libraries), where having an easy way to write test setup goes a long way for people to actually write useful tests.

Reply:
This laziness comes at a cost -- for example your current netns usage that executes two commands (one to create the network ns and another to `up` it) can fail to run the second half, and the Drop impl won't run in that case. I think we want to be more scrupulous here, not less.

Reply:
Indeed, good point. Though I still stand that for tests themselves this helper for running a script is useful.

Properly split with cleanups in f4c3968. It does so by creating the handle asap so that drop gets called if link up fails.

Reply:
I buy that we want to make this convenient for callers, but I don't think we should do that by using bash. Do we really want to end up with word splitting issues in our tests because we decided bash was an appropriate language?

Better would be to make this function's signature:
```rust
fn script<T: AsRef<OsStr>, I: IntoIterator<Item=T>>(script: impl IntoIterator<Item = I>)
```
note that this is different from my earlier suggestion in two ways:
- [u8] -> OsStr to match the bounds on various methods on `Command`
- `script` is now an iterator of iterators, so that this can be called as:
```rust
script([
  &["echo", "hello world"], // Look ma, no word splitting!
  &["rm", "-rf", "/"]
])
```

stepping back even further: we really really want to discourage shelling out in our tests because we want to drive toward having virtualized tests that exercise different kernel versions, and we don't want the userland of those VMs to become load bearing. We want to build the integration tests once, and then run them on a bunch of different VMs which ideally just have the kernel and little else.

Could we do this NetNS stuff using [the netns_rs crate](https://docs.rs/netns-rs/latest/netns_rs/) which doesn't depend on any userland at all?

Reply:
As requested:

- dropped the script function (well, not requested per se, but strongly implied)
- dropped the usage of iproute2, both for netns creation and for link up